#2824 Fixed event handlers new xid not displaying after saving edit form

Patrykb0802 opened the pull request.

Patrykb0802 committed:
#2824 Fixed event handlers new xid not displaying after saving edit form

Limraj commented on WebContent/WEB-INF/jsp/eventHandlers.jsp:
```diff
@@ -508,6 +508,7 @@
 
             selectedHandlerNode.object = handler;
         }
+        $set("xid", handler.xid);
```
Where is the definition of the handler variable?
What is defined in the else block is not available outside this block.
If the xid has not been completed when editing the handler, it should not save the newly generated xid and return the old xid. Otherwise, this is counterintuitive behavior and the user may unknowingly change the xid of the handler.

Patrykb0802 committed:
#2926 Fixed icons placement on data point list

- Added isNotEmpty() to common.js
- Moved validateValue() to common.js
- Added restoring previous XID value for Event Handlers when XID field is left empty when saving Event Handler

Patrykb0802 committed:
Revert "#2926 Fixed icons placement on data point list"

This reverts commit fa04fe79c7fb15a96f7694c0f0fa84b277325967.

Patrykb0802 committed:
#2824 Fixed event handlers new xid not displaying after saving edit form

- Added isNotEmpty() to common.js
- Moved validateValue() to common.js
- Added restoring previous XID value for Event Handlers when XID field is left empty when saving Event Handler

Limraj commented on WebContent/WEB-INF/jsp/eventHandlers.jsp:
```diff
@@ -456,36 +456,43 @@
+        var messages = [];
+        if (!(isNotEmpty(xid)) && handlerId !== ${NEW_ID}){
+          validateValue("xid", "<fmt:message key='validate.valueRestored'/>", isNotEmpty, xid, messages)
+          $set("xid", selectedHandlerNode.object.xid);
```
added lines:
```
stopImageFader("saveImg");
showDwrMessages(message);
return;
```

Limraj commented on WebContent/WEB-INF/jsp/eventHandlers.jsp:
```diff
@@ -456,36 +456,43 @@
+          validateValue("xid", "<fmt:message key='validate.valueRestored'/>", isNotEmpty, xid, messages)
+          $set("xid", selectedHandlerNode.object.xid);
+        }
+        if(messages.length > 0){
```
if to delete

Limraj commented on webapp-resources/messages_en.properties:
```diff
@@ -2989,6 +2989,7 @@ validate.required=Required value
 validate.text.incompatible=Text renderer is incompatible with data type
 validate.event.incompatible=Event text renderer is incompatible with data type
 validate.xidUsed=This XID is already in use
+validate.valueRestored=Previous value restored
```
Ok.
Need to update for each translation files.

Limraj commented on WebContent/WEB-INF/jsp/eventHandlers.jsp:
```diff
@@ -456,36 +456,43 @@
         var disabled = $get("disabled");
-        if (handlerType == <c:out value="<%= EventHandlerVO.TYPE_EMAIL %>"/>) {
+        var messages = [];
+        if (!(isNotEmpty(xid)) && handlerId !== ${NEW_ID}){
```
`!(isNotEmpty(xid))` - Instead of so many negations, it is worth creating the isEmpty method.

Limraj commented on WebContent/WEB-INF/jsp/eventHandlers.jsp:
```diff
@@ -456,36 +456,43 @@
-        if (handlerType == <c:out value="<%= EventHandlerVO.TYPE_EMAIL %>"/>) {
+        var messages = [];
+        if (!(isNotEmpty(xid)) && handlerId !== ${NEW_ID}){
+          validateValue("xid", "<fmt:message key='validate.valueRestored'/>", isNotEmpty, xid, messages)
```
Remove use validateValue:
let message = createValidationMessage("xid", "<fmt:message key='validate.valueRestored'/>");

Limraj commented on WebContent/WEB-INF/jsp/eventHandlers.jsp:
```diff
@@ -456,36 +456,43 @@
         var alias = $get("alias");
         var disabled = $get("disabled");
-        if (handlerType == <c:out value="<%= EventHandlerVO.TYPE_EMAIL %>"/>) {
+        var messages = [];
```
messages to delete

Limraj commented on src/com/serotonin/mango/web/dwr/EventHandlersDwr.java:
```diff
@@ -325,7 +325,12 @@ private DwrResponseI18n save(int eventSourceId, int eventTypeRef1,
 		vo.setId(handlerId);
-		vo.setXid(StringUtils.isEmpty(xid) ? eventService.generateUniqueXid() : xid);
+		if (StringUtils.isEmpty(xid) && handlerId == Common.NEW_ID) {
+			vo.setXid(StringUtils.isEmpty(xid) ? eventService.generateUniqueXid() : xid);
```
vo.setXid(eventService.generateUniqueXid());

Patrykb0802 committed:
#2824 Fixed event handlers new xid not displaying after saving edit form

Limraj commented on webapp-resources/messages_es.properties:
```diff
@@ -2948,6 +2948,7 @@ validate.required=Se requiere un valor
 validate.text.incompatible=El generador de Texto es incompatibles con el tipo de dato
 validate.event.incompatible=Event text renderer is incompatible with data type
 validate.xidUsed=Este XID ya est\u00e1 en uso
+validate.valueRestored=Previous value restored
```
You need to add this at the end of the file, not in the middle.

Limraj commented on WebContent/WEB-INF/jsp/eventHandlers.jsp:
```diff
@@ -456,36 +456,40 @@
+          let message = createValidationMessage("xid", "<fmt:message key='validate.valueRestored'/>");
+          $set("xid", selectedHandlerNode.object.xid);
+          stopImageFader("saveImg");
+          showDwrMessages([message]);
```
Add line:
return;

Limraj commented on WebContent/WEB-INF/jsp/eventHandlers.jsp:
```diff
@@ -456,36 +456,40 @@
+          stopImageFader("saveImg");
+          showDwrMessages([message]);
+        }
+        else {
```
What comes after ifa needs to be restored. 
You only add a new if at the end of return, we do not make any other changes.

Patrykb0802 committed:
#2824 Fixed event handlers new xid not displaying after saving edit form

Patrykb0802 committed:
#2824 Fixed event handlers new xid not displaying after saving edit form

- Code formatting adjustments

Patrykb0802 committed:
#2824 Fixed event handlers new xid not displaying after saving edit form

Patrykb0802 committed:
#2824 Fixed event handlers new xid not displaying after saving edit form

- Adjusted code formatting

Limraj reviewed (approved):
If we create a new handler and leave the xid field empty, a new handler is created with the generated xid, if we try to save an existing event handler with an empty xid field, we get the currently saved xid with information 'Previous value restored' -> ok.
